usrp_drivers/usrp_driver.cpp

```diff
@@ -273,8 +273,26 @@ void transmit(zmq::context_t &driver_c, USRP &usrp_d, const DriverOptions &drive
     // Getting usrp box time to find out when to send samples. box_time continuously being updated.
     auto delay = uhd::time_spec_t(SET_TIME_COMMAND_DELAY);
     auto time_now = borealis_clocks.box_time;
+    // Earliest possible time to start sending samples
     auto sequence_start_time = time_now + delay;
 
+    if (driver_packet.align_sequences() == true) {
+      // Get the digit of the next tenth of a second after min_start_time
+      double tenth_of_second = std::ceil(sequence_start_time.get_frac_secs() * 10);
+      double fractional_second = tenth_of_second / 10;
+      // this occurs if the current time is 0.9+ seconds, so the rounding takes it up to 1.0 seconds.
+      if (fractional_second >= 0.95) {
+        fractional_second = 0.0;
+      }
+
+      // Start the sequence at the next tenth of a second.
+      if (fractional_second < 0.05) {
```

Comment:
50 milliseconds magic number -> determined empirically?
actually, doesn't this only happen if the fractional second is 0? Since the ceil() will always eliminate all but one non-zero digit?

Reply:
Yes that's correct, I just put 0.05 since it falls between 0.0 and 0.1. Probably should have commented this better